Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -592,6 +602,138 @@ RefPtr<StyleRuleFontFace> CSSParserImpl::consumeFontFaceRule(CSSParserTokenRange
     return StyleRuleFontFace::create(createStyleProperties(m_parsedProperties, m_context.mode));
 }
 
+static std::pair<FontFeatureValuesType, unsigned> fontFeatureValuesTypeMappings(CSSAtRuleID id)
+{
+    switch (id) {
+    // Styleset has no maximum in theory, but in practise we don't consider more than 99 values.
+    case CSSAtRuleStyleset: return { FontFeatureValuesType::Styleset, 99 }; 
+    case CSSAtRuleStylistic: return { FontFeatureValuesType::Stylistic, 1 };
+    case CSSAtRuleCharacterVariant: return { FontFeatureValuesType::CharacterVariant, 2 };
+    case CSSAtRuleSwash: return { FontFeatureValuesType::Swash, 1 };
+    case CSSAtRuleOrnaments: return { FontFeatureValuesType::Ornaments, 1 };
+    case CSSAtRuleAnnotation: return { FontFeatureValuesType::Annotation, 1 };
+    default:
+        ASSERT_NOT_REACHED();
+        return { };
+    }
+}
+
+RefPtr<StyleRuleFontFeatureValuesBlock> CSSParserImpl::consumeFontFeatureValuesRuleBlock(CSSAtRuleID id, CSSParserTokenRange prelude, CSSParserTokenRange range)
```

Comment:
Both of these new rule parsing paths should call into m_observableWrapper->observer() so that Web Inspector's parsing observer doesn't end up with an incorrect count of rules. See `consumeFontPaletteValuesRule` for an example of what that looks like.

Reply:
I guess technically we don’t keep these rules around, but we should still make sure the observer gets a complete picture.

Reply:
I've tried to add support for the observer, but I would like to add a test to be sure that the Web Inspector is working properly. Do we have an example of such a test ?

Reply:
@patrickangle I've added the observer code, if you want to take a look at it.